.ci/new_ci.jenkinsfile

```diff
@@ -0,0 +1,435 @@
+            compilerImages.add(image)
+        })
+
+        clangVersions.each({ String clangVersion ->
+            String image = "${dockerUsername}/clang${clangVersion}"
+            compilerImages.add(image)
+        })
+
+        List<String> conanImages = []
+        conanVersions.each({ String conanVersion ->
+            compilerImages.each({ String compilerImage ->
+                conanImages.add("${compilerImage}:${conanVersion}")
+            })
+        })
+
+        List<String> allImages = conanImages.collect()
+        conanImages.each({String conanImage ->
+            def (name, conanVersion) = conanImage.split(':')
+            jenkinsAgentVersions.each({String jenkinsAgentVersion ->
+                allImages.add("${name}-jenkins:${conanVersion}-${jenkinsAgentVersion}")
+            })
+        })
+
+        allImages.each({ String image ->
+            Map imageData = ["buildArgs": [:]]
+            imageData["name"] = "${image}-#${pr_number}"
+            imageData["compiler"] = image.contains("gcc") ? "gcc" : "clang"
+            String compilerVersionSubStr = image.split(imageData["compiler"]).last()
+            imageData["compilerVersion"] = compilerVersionSubStr.contains("-") ? compilerVersionSubStr.split("-").first() : compilerVersionSubStr.split(":").first()
+            imageData["buildArgs"]["CONAN_VERSION"] = image.split(":").last().split("-").first()
+            imageData["path"] = "legacy/${imageData["compiler"]}_${imageData["compilerVersion"]}"
+
+            if (image.contains("jenkins")) {
+                imageData["buildArgs"]["SOURCE_CONANIO_IMAGE"] = "${registry}/${image.split("-").first()}:${imageData["buildArgs"]["CONAN_VERSION"]}-#${pr_number}"
+                imageData["buildArgs"]["AGENT_VERSION"] = image.split(":").last().split("-")[1]
+                imageData["buildArgs"]["JAVA_JDK_VERSION"] = envValues["JAVA_JDK_VERSION"]
+                imageData["path"] = "legacy/jenkins"
+            }
+            echo "${imageData}"
+            images.add(imageData)
+        })
+    }
+    return images
+}
+
+def getModernImages(String pr_number, String registry) {
+    Map<String, String> envValues = [:]
+    stage('Modern - Parse environment file') {
+        envValues = readEnvValues("modern/.env")
+    }
+
+    List<Map> images = []
+    stage('Modern - Generate build matrix') {
+        List<String> gccVersions = envValues["GCC_VERSIONS"].split(",").collect({String version -> return getVersionMajor(version)})
+        List<String> clangVersions = envValues["CLANG_VERSIONS"].split(",").collect({String version -> return getVersionMajor(version)})
+        List<String> conanVersions = [envValues["CONAN_VERSION"]]
+        List<String> jenkinsAgentVersions = envValues["JENKINS_AGENT_VERSION"].split(",")
+        String cmakeVersion = envValues["CMAKE_VERSION_FULL"]
+        String pythonVersion = envValues["PYTHON_VERSION"]
+        String dockerUsername = envValues["DOCKER_USERNAME"]
```

Comment:
We could update in another PR, but DOCKER_USERNAME is wrong, because it reflects to the login. What we are actually is referring it the repository , like conan-io/gcc5, conan-io the is repository. So it should be DOCKER_REPOSITORY. What comes before, like Artifactory address, is names as namespace. 

https://docs.docker.com/reference/cli/docker/image/tag/